scripts/data_prep/convert_text_to_mds.py

```diff
@@ -0,0 +1,352 @@
+# Copyright 2022 MosaicML LLM Foundry authors
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+import math
+import os
+import tempfile
+from argparse import ArgumentParser, Namespace
+from concurrent.futures import ProcessPoolExecutor
+from glob import glob
+from typing import Iterable, List, Tuple, cast
+
+from composer.utils import (ObjectStore, maybe_create_object_store_from_uri,
+                            parse_uri)
+from streaming import MDSWriter
+from tqdm import tqdm
+from transformers import AutoTokenizer
+
+from llmfoundry.data import ConcatTokensDataset
+from llmfoundry.utils.data_prep_utils import (DownloadingIterable,
+                                              merge_shard_groups)
+
+
+def parse_args() -> Namespace:
+    """Parse commandline arguments."""
+    parser = ArgumentParser(
+        description=
+        'Convert text files into MDS format, optionally concatenating and tokenizing'
+    )
+    parser.add_argument(
+        '--max_mds_writer_workers',
+        type=int,
+        default=64,
+        help='The maximum number of workers to use for MDS writing')
+    parser.add_argument('--output_folder',
+                        type=str,
+                        required=True,
+                        help='The folder to write output to')
+    parser.add_argument('--input_folder',
+                        type=str,
+                        required=True,
+                        help='The folder with text files to convert to mds')
+    parser.add_argument('--compression',
+                        type=str,
+                        default='zstd',
+                        help='The compression algorithm to use for MDS writing')
+
+    group = parser.add_mutually_exclusive_group(required=False)
+    group.add_argument(
```

Comment:
I think this group thing is no longer correct. `--concat_tokens` is a required arg for this script, since this script doesn't support the no concat mode.